src/test/java/org/broadinstitute/hellbender/tools/walkers/annotator/VariantAnnotatorEngineUnitTest.java

```diff
@@ -0,0 +1,149 @@
+    public void testBadGroup(){
+        final List<String> annotationsToExclude= Collections.emptyList();
+        final FeatureInput<VariantContext> dbSNPBinding = null;
+        final List<String> annotationGroupsToUse = Arrays.asList("fred");
+        final List<String> annotationsToUse = Arrays.asList(Coverage.class.getSimpleName());//good one
+        final VariantAnnotatorEngine vae = VariantAnnotatorEngine.ofSelected(annotationGroupsToUse, annotationsToUse, annotationsToExclude, dbSNPBinding);
+        Assert.assertFalse(vae.getGenotypeAnnotations().isEmpty());
+        Assert.assertFalse(vae.getInfoAnnotations().isEmpty());
+    }
+
+
+    private VariantContext makeVC(final Allele refAllele, final Allele altAllele) {
+        final List<Allele> alleles = Arrays.asList(refAllele, altAllele);
+        final Genotype g = new GenotypeBuilder("sample1", alleles).make();
+
+        return (new VariantContextBuilder())
+                .alleles(Arrays.asList(refAllele, altAllele)).chr("1").start(15L).stop(15L).genotypes(g).make();
+    }
+
+    private Map<String, PerReadAlleleLikelihoodMap> makeReadMap(final int ref, final int alt, final Allele refAllele, final Allele altAllele) {
+        final PerReadAlleleLikelihoodMap map= new PerReadAlleleLikelihoodMap();
+
+        for (int i = 0; i < alt; i++) {
+            final GATKRead r = ArtificialReadUtils.createArtificialRead(TextCigarCodec.decode("10M"));
+            map.add(r, altAllele, -1.0);
+            map.add(r, refAllele, -100.0);
+        }
+        for (int i = 0; i < ref; i++) {
+            final GATKRead read = ArtificialReadUtils.createArtificialRead(TextCigarCodec.decode("10M"));
+            map.add(read, altAllele, -100.0);
+            map.add(read, refAllele, -1.0);
+        }
+
+        return Collections.singletonMap("sample1", map);
+    }
+
+    @Test
+    public void testCoverageAnnotation() throws Exception {
+        final List<String> annotationGroupsToUse= Collections.emptyList();
+        final List<String> annotationsToUse = Arrays.asList(Coverage.class.getSimpleName());//good one
+        final List<String> annotationsToExclude= Collections.emptyList();
+        final FeatureInput<VariantContext> dbSNPBinding = null;
+        final VariantAnnotatorEngine vae = VariantAnnotatorEngine.ofSelected(annotationGroupsToUse, annotationsToUse, annotationsToExclude, dbSNPBinding);
+
+        final int alt = 5;
+        final int ref = 3;
+        final Allele refAllele = Allele.create("A", true);
+        final Allele altAllele = Allele.create("T");
+
+        final Map<String, PerReadAlleleLikelihoodMap> perReadAlleleLikelihoodMap = makeReadMap(ref, alt, refAllele, altAllele);
+        final VariantContext resultVC = vae.annotateContext(new FeatureContext(), null, makeVC(refAllele, altAllele), perReadAlleleLikelihoodMap, a->true);
+        Assert.assertEquals(resultVC.getCommonInfo().getAttribute(VCFConstants.DEPTH_KEY), String.valueOf(ref+alt));
+    }
```

Comment:
Need more and better tests for `annotateContext()` (multiple annotations at once, overlapping features in the `FeatureContext`, non-null DBSNP input). Also need individual tests for `annotateGenotypes()`